javascript/ql/src/semmle/javascript/security/dataflow/RegExpInjectionCustomizations.qll

```diff
@@ -55,4 +55,27 @@ module RegExpInjection {
       )
     }
   }
+
+  /**
+   * A global regexp replacement involving the `{`, `[`, or `+` meta-character, viewed as a sanitizer for
+   * regexp-injection vulnerabilities.
+   */
+  class MetacharEscapeSanitizer extends Sanitizer, StringReplaceCall {
+    MetacharEscapeSanitizer() {
+      isGlobal() and
+      (
+        RegExp::alwaysMatchesMetaCharacter(getRegExp().getRoot(), ["{", "[", "+"])
+        or
+        // or it's like a wild-card.
+        RegExp::isWildcardLike(getRegExp().getRoot())
+      )
+    }
+  }
+
+  /**
+   * Meta characters used in the above sanitizer.
+   */
+  private class RegexpMetaChars extends RegExp::MetaCharacter {
```

Comment:
Could you make sure this is (privately) imported by default, or moved into `RegExp::`? I'd rather avoid re-evaluating `alwaysMatchesMetaCharacter` if it's just a matter of adding the right imports.